### PR TITLE
TODO-LIST(adding and removing)

### PR DESCRIPTION
**Pull request by elyor-doniyorov**
### Add, Remove and Edit functionalities

**In this milestone, I:👇**

- Create add, remove and edit functions;
- Use Webpack and ES6;
- Use HTML, CSS, Javascript linter configs:
- Use GitHub flow;

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -37,14 +36,54 @@ const createList = (todoItem) => {
...
+
+    tasks.splice(id, 1);
+    for (let i = 0; i < tasks.length; i += 1) {
+      tasks[i].index = i;
+    }
+
+    localStorage.setItem('items', JSON.stringify(tasks));
+    location.reload();
+  }));
+};
```
Comment (bellom) [inline src/index.js]: Let's move this to a new file just as we did for `addTodo` and `deleteAll`

File: src/index.js [hunk 2]
```diff
@@ -37,14 +36,54 @@ const createList = (todoItem) => {
...
+
+  for (let i = 0; i < todoTextSpan.length; i += 1) {
+    todoTextSpan[i].addEventListener('keydown', (event) => {
+      if (event.key === 'Enter') {
+        tasks[i].description = todoTextSpan[i].textContent;
+        localStorage.setItem('items', JSON.stringify(tasks));
+      }
+    });
+  }
+};
```
Comment (bellom) [inline src/index.js]: Let's move this to a new file just as we did for `addTodo` and `deleteAll`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @elyor-doniyorov  👏 , 

### Awesome job so far!! :+1: 

#### You have done wonderfully well creating add, remove and edit function, setting up linter and good commit messages 🔥, However, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Required Changes! ♻️ 

_1. Kindly check the comments under the review._


### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bellom — COMMENTED** [review]
### [Optional] Sorry this came in late, I thought the issue came from PC, after couples mins trying to debug so I decided to use another PC, still the same 🤦, it would be nice to fix this.

Kindly fix the npm start command.

We are using Webpack to run this project, our webpack dev server should be working properly, users should be able to run npm start command successfully, in your case for some reason, it is not.

File: src/index.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
+/* eslint-disable no-restricted-globals, prefer-destructuring, import/no-unresolved */
 import './style.css';
 import todoStatusUpdate from './todoStatusUpdate.js';
 import dragDrop from './dragDrop.js';
+import deleteCompletedTodoItem from './deleteAll.js';
+import addTodo from './addTodo.js';
+import edit from './editTask.js';
```
Comment (mimipeshy) [inline src/index.js]: - [ ] There is this error ⬇️ due to a typo , its looking for a file named `./addTodo.js` but this is what exists `/addToDo.js`. Kindly fix this
Screenshot from 2021-08-20 11-34-13

**Review by mimipeshy — CHANGES_REQUESTED** [review]
## Hi @elyor-doniyorov 👋🏾 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️

- [ ] I also got the same error the previous was taking about when running `npm start`. In webpack.config file line 8, I changed it from `contentBase` to `static` and it worked. You [can read more about the error here](https://stackoverflow.com/questions/67926476/webpack-dev-server-config-contentbase-not-working-in-latest-version). You can consider working with this.

- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](<https://microverse.zendesk.com/hc/en-us/articles/1500004088561>)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](<https://airtable.com/shrQAqnBwek5a0O0s>)._

**Review by hemantso — APPROVED** [review]
Hi @elyor-doniyorov , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**